platform/m6/lib.c

```diff
@@ -0,0 +1,142 @@
+#include "platform.h"
+#include "lolevel.h"
+#include "live_view.h"
+
+#include "lib.h"
+
+void shutdown_soft() {
+	_PostLogicalEventForNotPowerType(0x1005, 0);
+}
+
+/**
+ * @see main startup
+ */
+void shutdown() {
+	while (1)
+		_SleepTask(500);
+}
+
+/**
+ * @header debug_led.h
+ */
+void debug_led(int state) {
+	volatile long *p = (void*) LED;
+	*p = ((state) ? 0x4d0002 : 0x4c0003);
+}
+
+/**
+ * Power Led = first entry in table (led 0).
+ * AF Assist Lamp = second entry in table (led 1).
+ *
+ * @header debug_led.h
+ */
+void camera_set_led(int led, int state, __attribute__ ((unused))int bright) {
+	static char led_table[2] = { 0, 4 };
+	_LEDDrive(led_table[led % sizeof(led_table)], state <= 1 ? !state : state);
+}
+
+/**
+ * @header keyboard.h
+ */
+void JogDial_CW(void) {
+	_PostLogicalEventToUI(0x872, 1); // RotateJogDialRight
+}
+
+/**
+ * @header keyboard.h
+ */
+void JogDial_CCW(void) {
+	_PostLogicalEventToUI(0x873, 1); // RotateJogDialLeft
+}
+
+/**
+ * @header properties.h
+ */
+int get_flash_params_count(void) {
+	return 0x113; // see j_PTM_BackupUIProperty_FW
+}
+
+/**
+ * @header raw_buffer.h
+ */
+char* hook_raw_image_addr() {
+	if (current_raw_addr) {
+		return current_raw_addr;
+	}
+	return (char*) 0x48c57740; // TODO
```

Comment:
not sure about this ?

Reply:
I think I got that address by logging the raw buffers address used while shooting. It's used as fallback, for things like the benchmark module.